packages/app/src/vis-packs/nexus/utils.ts

```diff
@@ -147,7 +148,6 @@ export function findAssociatedDatasets(
     assertDefined(dataset, `Expected child entity "${name}" to exist`);
     assertDataset(dataset, `Expected child "${name}" to be a dataset`);
     assertArrayShape(dataset);
-    assertNumericType(dataset);
```

Comment:
I struggled with type-checking a bit because of this type assertion and because `axes` datasets are still expected to have a numeric type, not a complex type. I considered asserting for numeric or complex type here conditionally based on the `type` parameter, but this felt like a really bad idea.

In the end, it was much easier to remove the assertion from here and to move it inside `findAxesDatasets` instead. I then created a new `findAuxiliaryDatasets` function that asserts for numeric or complex type.

At the end we still get `NxData<NumericType | ComplexType>` (but now with both the signal and auxiliaries of type `Dataset<ArrayShape, NumericType | ComplexType>`), and we narrow that to just `NumericType` or just `ComplexType` with `assertNumericNxData` or `assertComplexNxData`.